deeptime/src/include/deeptime/markov/msm/tram/tram.h

```diff
@@ -146,13 +163,16 @@ struct TRAM {
     // returned through the callback function if it is provided. log-likelihoods are returned if
     // trackLogLikelihoods = true. By default, this is false, since calculating the log-likelihood required
     // an estimation of the transition matrices, which slows down estimation.
-    void estimate(std::size_t maxIter, dtype maxErr, bool trackLogLikelihoods = false,
+    void estimate(std::shared_ptr<TRAMInput<dtype>> &tramInput, std::size_t maxIter, dtype maxErr,
+                  std::size_t callbackInterval = 1, bool trackLogLikelihoods = false,
                   const py::object *callback = nullptr) {
 
-        // Array to keep track of _statVectors(K, i) = exp(_thermStateEnergies(K) - _biasedConfEnergies(K, i))
-        // difference between previous and current statvectors is used for calculating the iteration error.
-        auto statVectors = ExchangeableArray<dtype, 2>(std::vector({nThermStates_, nMarkovStates_}), 0.);
-        py::gil_scoped_release gilRelease;
+        input_ = tramInput;
+
+        if (initializeParametersOnEstimationStart_) {
+            initLagrangianMult();
```

Comment:
is this something that can (and should) be done in the constructor? I like to think of it as something that is called and then the object is ready to use

Reply:
initLagrangianMult() needs the TRAMInput to run. I didn't want to construct the estimator with the data as a parameter because I think the data should be passed into estimate(). I agree this is not a very nice solution. But I don't know a better way at the moment.

Reply:
I got rid of the boolean at least, as it was unnessecary